app/scripts/article/config.js

```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+module.exports = [
+    '$stateProvider',
+    function articleConfig($stateProvider) {
+
+        $stateProvider
+          .state('article', {
+              url          : "/article/:id",
+              templateUrl  : 'scripts/article/views/article.html',
+              controller   : 'articleController'
```

Comment:
Add a `resolve` callback here and try to package everything article-related before returning the article collection back to the controller.